### DISCUSSION TITLE
Login returnUrl with OpenIdConnect Client

### DISCUSSION THREAD

**Discussion by snovak7** [OrchardCMS/OrchardCore#7394] [Q&A]
Hello, I need some guidance with returnUrl... I have this requirement that certain Url is a protected area (authorization required). When the user clicks on the link, it redirects to that IdP, and when it returns that returnUrl is lost, I know that OIDC in asp.net core has this /signin-oidc URL baked in, but somehow there must be a way to achieve this?

Once the user is Authorized, all is working as expected. (on the 2nd try)

**Comment by Skrypt** [comment]
As far as I know, the Login page can use a returnUrl. I don't know details about OpenId though.

**Reply by snovak7** [reply]
Well, I guess if I copy all the code to my own module, it seems to work ok, I removed the first "if" which deals if the user is logged in strip out the return URL.
